src/parser/de.rs

```diff
@@ -0,0 +1,309 @@
+  for char in qry.chars() {
+    match char {
+      ' ' => (),
+      '-' => {
+        b = true;
+      }
+      _ => {
+        if b {
+          s.push(char.to_ascii_uppercase());
+        } else {
+          s.push(char);
+        }
+        b = false;
+      }
+    }
+  }
+  s
+}
+
+fn to_json(value: &Value, result: &mut HashMap<usize, PosValHolder>, prl: (Option<String>, &String, usize)) {
+  match value {
+    Value::Null | Value::Bool(_) | Value::Number(_) => (),
+    Value::String(s) => {
+      let (parent_key, root_node, level) = prl;
+      let y = parent_key.unwrap_or_default();
+      let v = result.entry(level).or_default().entry(root_node.clone()).or_default();
+      if !y.is_empty() {
+        v.push((y, s.clone()));
+      }
+    }
+    Value::Array(arr) => {
+      let (parent_key, root_node, level) = prl;
+      for v in arr.iter() {
+        to_json(v, result, (parent_key.clone(), root_node, level + 1));
+      }
+    }
+    Value::Object(obj) => {
+      let (_, root_node, level) = prl;
+      for (k, v) in obj.iter() {
+        to_json(
+          v,
+          result,
+          (
+            Some(k.to_string()),
+            if v.is_object() { k } else { root_node },
+            level + 1,
+          ),
+        );
+      }
+    }
+  }
+}
+
+fn to_json_str(value: &Value) -> String {
+  match value {
+    Value::Null => "".to_string(), // Return empty string for null values
+    Value::Bool(b) => b.to_string(),
+    Value::Number(num) => num.to_string(),
+    Value::String(s) => s.to_string(),
+    Value::Array(arr) => {
+      let elements: Vec<String> = arr.iter().map(to_json_str).collect();
+      format!("[{}]", elements.join(" "))
+    }
+    Value::Object(obj) => {
+      let pairs: Vec<String> = obj.iter().map(|(k, v)| get_cur(k, v)).collect();
+      format!("{{{}}}", pairs.join(" "))
+    }
+  }
+}
+
+fn next_token(input: &str, position: &mut usize) -> Option<char> {
+  if let Some(ch) = input.chars().nth(*position) {
+    *position += 1;
+    Some(ch)
+  } else {
+    None
+  }
+}
+
+fn get_cur(k: &String, v: &Value) -> String {
+  let s = to_json_str(v);
+  if s.is_empty() {
+    k.clone()
+  } else {
+    format!("{} {}", k, s)
+  }
+}
+
+#[cfg(test)]
+mod parser_tests {
+  use crate::async_graphql_hyper::GraphQLRequest;
+  use crate::parser::de::Parser;
+
+  #[test]
+  fn t1_url_qry_parser() {
+    let parser = Parser::from_path("api/user?$=name,age,address.city,address.state");
+    assert_eq!(
+      parser.unwrap().parse::<GraphQLRequest>().unwrap().0.query,
+      "{user {address {city state} age name}}"
+    );
+  }
+  #[test]
+  fn t2_url_nested_qry_parser() {
+    let parser =
+      Parser::from_path("api/user?id=123,address.country=India,address.city=Foo&$=name,age,address.city,address.state");
+    assert_eq!(
+      parser.unwrap().parse::<GraphQLRequest>().unwrap().0.query,
+      "{user (id: 123,) {address (city: Foo,country: India,) ) {city state} age name}}"
+    );
+  }
+}
```

Comment:
Add support for `*` also which expands on all the fields, refer to the original issue for more details.